### PR TITLE
Fixed coral sell & added allowSwimming Ox_lib boolean to progressbar

### PR DESCRIPTION
**Pull request by IN1GHTM4R3**
## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Comment by mafewtm** [comment]
Instead of instructing the user to comment out a line of ox_lib, we should just be adding `allowSwimming = true` to the progress bars. Additionally, you don't need to add a comment in the code explaining why its being done. That's what the description of this PR is for

**Comment by IN1GHTM4R3** [comment]
Ok @mafewtm should be fixed now based on the above.

File: server/main.lua [hunk 1]
```diff
@@ -40,7 +40,7 @@ RegisterNetEvent('qbx_diving:server:sellCoral', function()
     local src = source
     local player = exports.qbx_core:GetPlayer(src)
     local availableCoral = getCoralInInventory(src)
-    if #availableCoral == 0 then
+    if availableCoral == 0 then
```
Comment (Manason) [inline server/main.lua]: This won't work as available Coral is a table. Can use table.next to check for an empty table

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (IN1GHTM4R3) [reply server/main.lua]: @Manason I unfortunately don't know how to use table.next if you have documentation on how to properly implement it. I have googled it and don't seem to find much on it or im just not seeing how.

Comment (mafewtm) [reply server/main.lua]: 6.1 - Basic Functions of https://www.lua.org/manual/5.4/manual.html

In this current case I believe he wants you to do `if not next(availableCoral)` which checks if there are no more key-value pairs to iterate over. He can correct me if I'm mistaken

File: client/main.lua [hunk 2]
```diff
@@ -18,6 +18,7 @@ local function takeCoral(coralIndex)
         label = Lang:t('info.collecting_coral'),
         canCancel = true,
         useWhileDead = false,
+        allowSwimming = true,
```
Comment (Manason) [inline client/main.lua:21]: At the top of server/main.lua add a dependency check to assert that the server has the minimum ox_lib version for this field https://overextended.dev/ox_lib/Modules/Version/Shared

**Review by Manason — APPROVED** [review]

**Review by mafewtm — APPROVED** [review]

**Pull request merged**